toolbox/timecomparer/main.go

```diff
@@ -0,0 +1,537 @@
+			runTimes:  undefinedRuntimes,
+		}
+		commandToTime[command] = rCombination
+
+		commandMap[command] = sections[2:]
+	}
+
+	return commandMap, commandToTime, nil
+}
+
+func processCommand(commandToTime map[string]runCombination) ([]float64, map[float64]commandAndTime) {
+	var timeArray []float64
+	commandsBreakDown := map[float64]commandAndTime{}
+	for command, rCombination := range commandToTime {
+		time := rCombination.totalTime
+		rTimes := rCombination.runTimes
+		averageTime := time / float64(len(rTimes))
+		cAndTime := commandAndTime{
+			command:    command,
+			runStrings: []string{},
+		}
+		for _, r := range rTimes {
+			cAndTime.runStrings = append(cAndTime.runStrings, r.toStringSlice()...)
+		}
+		timeArray = append(timeArray, averageTime)
+		commandsBreakDown[averageTime] = cAndTime
+	}
+	return timeArray, commandsBreakDown
+}
+
+// Convert minutes and sesconds ("0m1.061s") to float object.
+func convertStringToTime(timeString string) ([]float64, error) {
+	var timeSlice []float64
+	if strings.Contains(timeString, "m") {
+		timeSplit := strings.Split(timeString, "m")
+		minString := timeSplit[0]
+		fmt.Println("get minutes")
+		fmt.Println(minString)
+		minInt, err := strconv.Atoi(minString)
+		if err != nil {
+			fmt.Println("error parsing float minute")
+			timeSlice = append(timeSlice, -1.0)
+		} else {
+			minFloat := float64(minInt)
+			timeSlice = append(timeSlice, minFloat)
+		}
+		if len(timeSplit) > 1 {
+			secString := timeSplit[len(timeSplit)-1]
+			secValueString := strings.Split(secString, "s")[0]
+			fmt.Println("get seconds")
+			fmt.Println(secValueString)
+			secFloat, err := strconv.ParseFloat(secValueString, 64)
+			if err != nil {
+				return timeSlice, fmt.Errorf("error parsing float second")
+			}
+			timeSlice = append(timeSlice, secFloat)
+		}
+	}
+	return timeSlice, nil
+}
+
+// Extract time from runtime string with the format of "real	0m1.061s"
+func extractTime(runTimeString string) runPair {
+	timeSplit := strings.Fields(runTimeString)
+	length := len(timeSplit)
+	var ind int
+	for ind = length - 1; ind > 0; ind-- {
+		if strings.Compare(timeSplit[ind], "") != 0 {
+			break
+		}
+	}
+	timePart := timeSplit[ind]
+	timeSlice, _ := convertStringToTime(timePart)
+
+	for ind = 0; ind < length; ind++ {
+		if strings.Compare(timeSplit[ind], "") != 0 {
+			break
+		}
+	}
+	namePart := timeSplit[ind]
+
+	rPair := runPair{
+		name:       namePart,
+		time:       timeSlice,
+		stringTime: timePart,
+	}
+
+	return rPair
+
+}
+
+// Compute average user/real/sys time to compare with others.
+func toSeconds(realTime runPair) float64 {
+	realT := realTime.time
+	seconds := realT[0]*60 + realT[1]
+
+	return seconds
+}
+
+// Get error contents of files to update command-time map with commands and path to build logs.
+func (tc *TimeComparer) findTimeCommands(
+	ctx context.Context, filePaths []string, commandToTime map[string]runCombination) map[string]runCombination {
+	timeCommand := regexp.MustCompile(`^time (.*);`)
```

Comment:
Consider defining these vars outside the function, so we can reuse them between calls